### PR TITLE
Changes for polls PR

### PR DESCRIPTION
**Pull request by DubyaDude**
Changes for https://github.com/DSharpPlus/DSharpPlus/pull/1813 that implements [polls](https://github.com/discord/discord-api-docs/pull/6746)

- Fixed using/namespace ordering.
- Removed DotSettings file.
- Added DotSettings to gitignore.
- Added polls to DiscordMessage object.
- Fix Discord Message Builder throwing an exception when the message only has a poll.

### CODE REVIEW

**Comment by DubyaDude** [comment]
Ran into an issue when trying to send a poll as the bot, not sure how to solve this/ran out of time for today. Feel free to pull if someone beats me to it in the main PR:
```csharp
var poll = new DiscordPollBuilder().WithQuestion("Question!").AddOption("A").AddOption("B").Build();
var message = new DiscordMessageBuilder().WithPoll(poll);
await ctx.Channel.SendMessageAsync(message);
```
```json
{
  "message": "Invalid Form Body",
  "code": 50035,
  "errors": {
    "poll": {
      "duration": {
        "_errors": [
          {
            "code": "NUMBER_TYPE_COERCE",
            "message": "Value \"2024-04-10T01:12:45.2068776+00:00\" is not int32."
          }
        ]
      }
    }
  }
}
```

**Comment by OoLunar** [comment]
If I'm reading DDocs correctly:
> `| duration | integer | Number of hours the poll should be open for, up to 7 days |`

We should be passing `DateTimeOffset.TotalHours` instead of the `DateTimeOffset` object itself

**Comment by DubyaDude** [comment]
I see, yea that'll need some changes in the builder then. Hopefully when I'm free. As said before, feel free to merge if someone else gets to it before me in the main PR.

**Comment by DubyaDude** [comment]
I no longer get Payload errors, now receiving 403s, likely because my bot doesn't have access.
So, untested atm.

File: DSharpPlus/DiscordIntents.cs
Comment (DubyaDude) [inline DSharpPlus/DiscordIntents.cs:189]: Upon adding the [documented intents](https://github.com/discord/discord-api-docs/pull/6746), currently receiving `Connection terminated (4013, 'Invalid intent(s).')`, might not be rolled out yet? Commented out for the time being for testing.
```
GUILD_MESSAGE_POLLS (1 << 24)
  - MESSAGE_POLL_VOTE_ADD
  - MESSAGE_POLL_VOTE_REMOVE
DIRECT_MESSAGE_POLLS (1 << 25)
  - MESSAGE_POLL_VOTE_ADD
  - MESSAGE_POLL_VOTE_REMOVE
```

File: DSharpPlus/Entities/Channel/Message/DiscordMessageBuilder.cs
Comment (Plerx2493) [inline DSharpPlus/Entities/Channel/Message/DiscordMessageBuilder.cs:100]: Please use a fully qualified using at the top of the file

File: DSharpPlus/DiscordIntents.cs
Comment (DubyaDude) [reply DSharpPlus/DiscordIntents.cs:189]: It was supposed to be deployed today, but still hasn't, will try again tomorrow.

**Comment by akiraveliara** [comment]
please merge master into this

**Comment by OoLunar** [comment]
I'm gonna break the commit history

File: DSharpPlus/Entities/Channel/Message/DiscordMessage.cs
Comment (DubyaDude) [inline DSharpPlus/Entities/Channel/Message/DiscordMessage.cs:745]: Unsure if we should take in an answerId here or a DiscordPollAnswer object.

Comment (Plerx2493) [reply DSharpPlus/Entities/Channel/Message/DiscordMessage.cs:744]: I think id is more convenient because its a simple index, you could provide an overload if you want

Comment (DubyaDude) [reply DSharpPlus/Entities/Channel/Message/DiscordMessage.cs:744]: [It's not guaranteed to stay an index for the future](https://github.com/discord/discord-api-docs/pull/6746/files#diff-81a2c7d9f047834aa38e226fa4c7bf74b98831cd65116f76ea5281b21a9ce759R71-R73), but yea I'll ad an override.

Comment (Plerx2493) [reply DSharpPlus/Entities/Channel/Message/DiscordMessage.cs:744]: Yeah but even if they would make it an abitray id its easier to store it in a db for example, also i like the "we provide methods with the least required parameters" way

**Comment by OoLunar** [comment]
> > I no longer get Payload errors, now receiving 403s, likely because my bot doesn't have access. So, untested atm.
> 
> ~just pass `X-Super-Properties` in request headers👍~ does not requires xsp header now

As much as we appreciate your effort to help, DSharpPlus implements everything through Discord Docs. If it's not documented or an open PR on their repo, we can't/won't use it.

**Comment by DubyaDude** [comment]
I didn't write it here, but they lifted the bot restriction a day or two after I wrote that, so no longer an issue anyway.

Docs are pulled. Will do a final round of testing hopefully after work today, otherwise this PR should be generally good to go, so if there are any more requested changes, go ahead and request them 👍

File: DSharpPlus/Entities/BaseDiscordMessageBuilder.cs
Comment (akiraveliara) [inline DSharpPlus/Entities/BaseDiscordMessageBuilder.cs:48]: can we take a DiscordPollBuilder here instead? i'm not a fan of exposing the REST payload type

File: DSharpPlus/Entities/Channel/Message/Poll/DiscordPollBuilder.cs
Comment (akiraveliara) [inline DSharpPlus/Entities/Channel/Message/Poll/DiscordPollBuilder.cs:32]: is there any reason we're using an integer and not a TimeSpan?

File: DSharpPlus/Entities/Channel/Message/Poll/DiscordPollBuilder.cs
Comment (akiraveliara) [inline DSharpPlus/Entities/Channel/Message/Poll/DiscordPollBuilder.cs:100]: i would prefer to not expose the payload type if possible

File: DSharpPlus/Entities/DiscordPermissions.cs
Comment (akiraveliara) [inline DSharpPlus/Entities/DiscordPermissions.cs:354]: update FULL_PERMS at the top of the file too

File: DSharpPlus/Entities/Interaction/DiscordInteractionApplicationCommandCallbackData.cs
Comment (akiraveliara) [inline DSharpPlus/Entities/Interaction/DiscordInteractionApplicationCommandCallbackData.cs:37]: can we take a DiscordPoll here instead?

File: DSharpPlus/Net/Rest/DiscordApiClient.cs
Comment (akiraveliara) [inline DSharpPlus/Net/Rest/DiscordApiClient.cs:5202]: for consistency, add `expire` to Endpoints.cs too

**Review by akiraveliara — CHANGES_REQUESTED** [review]

File: DSharpPlus/Entities/Channel/Message/Poll/DiscordPollBuilder.cs
Comment (DubyaDude) [reply DSharpPlus/Entities/Channel/Message/Poll/DiscordPollBuilder.cs:32]: Discord takes an integer called 'Duration' that represents the amount of hours the poll is up when creating the poll.
I could turn this into a TimeSpan if that's preferred.

File: DSharpPlus/Entities/BaseDiscordMessageBuilder.cs
Comment (akiraveliara) [reply DSharpPlus/Entities/BaseDiscordMessageBuilder.cs:48]: or a DiscordPoll, for that matter

File: DSharpPlus/Entities/Interaction/DiscordInteractionApplicationCommandCallbackData.cs
Comment (DubyaDude) [reply DSharpPlus/Entities/Interaction/DiscordInteractionApplicationCommandCallbackData.cs:37]: The class `DiscordInteractionApplicationCommandCallbackData` is internal here, and used for building request payloads. This one cannot be changed in the style requested.

File: DSharpPlus/Entities/Channel/Message/Poll/DiscordPollBuilder.cs
Comment (DubyaDude) [reply DSharpPlus/Entities/Channel/Message/Poll/DiscordPollBuilder.cs:32]: Is this still something you'd want to change? It doesn't make sense to make it a TimeSpan to me if discord will only accept an int that represents hours.

**Review by akiraveliara — APPROVED** [review]

**Pull request merged**
